backend/utils/controllers-utils/collections.ts

```diff
@@ -0,0 +1,63 @@
+export const findSearchText = (searchText: any, subcategories: any) => {
+    if(subcategories !== undefined) {
+        for(const category of subcategories){
+            for(const value of category.values) {
+                if(value.toString().includes(searchText)) {
+                    return true
+                }
+            }
+            if(findSearchText(searchText, category.subcategories)) {
+                return true
+            }
+        }
+    }
+    return false
+}
```

Comment:
This would be more readable I think
```suggestion
export const findSearchText = (searchText: string, subcategories: { values: any[], subcategories: any[] }[]): boolean => {
  if (!subcategories) return false;

  return subcategories.some(category => 
    category.values.some(value => value.toString().includes(searchText)) ||
    findSearchText(searchText, category.subcategories)
  );
};
```

Reply:
Below searches also can be refactored in the similar manner. Cuz n^2 (two loops) complexity for this ones is very heavy...

Reply:
Look also at `backend/utils/controllers-utils/data-export.ts` file there is the same situation ;)